deeppavlov/core/data/data_fitting_iterator.py

```diff
@@ -0,0 +1,68 @@
+"""
+Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from random import Random
+from typing import List, Generator, Tuple, Any
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.common.log import get_logger
+
+logger = get_logger(__name__)
+
+
+@register('data_fitting_iterator')
+class DataFittingIterator:
+    """
+    Dataset iterator for fitting models, e. g. vocabs, kNN, vectorizers.
+    Data is passed as a list of strings(documents).
+    Generate batches (for large datasets).
+    """
+
+    def __init__(self, data: List[str], doc_ids: List[Any]=None,
+                 seed: int = None, shuffle: bool = None, batch_size: int = None,
+                 *args, **kwargs) -> None:
+
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.random = Random(seed)
+        self.data = data
+        self.doc_index = doc_ids or self.get_doc_ids()
+        self.data = data
+
+    def get_doc_ids(self):
+        ids = [i for i in range(len(self.data))]
+        return ids
+
+    def gen_batch(self, batch_size=1000, shuffle=False) -> Generator[Tuple[List[list],
+                                                                           List[int]], Any, None]:
+        _batch_size = self.batch_size or batch_size
+        _shuffle = self.shuffle or shuffle
+
+        if _shuffle:
+            self.random.shuffle(self.doc_index)
```

Comment:
After getting shuffled data once, you loose the ability to get the data in original order.

Reply:
Why would I need the original order?

Reply:
Why do you need shuffle flag if you don't need original order?

Reply:
Original order can be used to properly compare different ways of processing. You could train your model on shuffled data and then predict on ordered, for example.

Reply:
If you always want to shuffle, you do not need a flag for it, you can just always shuffle.

Reply:
To compare different ways of processing you can shuffle with the same seed. 
" You could train your model on shuffled data and then predict on ordered, for example." - so set shuffle to True when training and set it to False when predicting?

Reply:
> so set shuffle to True when training and set it to False when predicting?   

Exactly. Preferably without the need to recreate the object.

Reply:
For this use case just leave the __init__ value as `None` and set it to `True` or `False` when shuffling for training and predicting accordingly.
The logic is clear, there are 2 possible use cases:
1. You want to preserve any argument value in a class instance, init it with the value you need and don't worry about messing up with different values (it does not always perform well). 
2. You want to change arguments. Then leave the default value as ` None` and call methods with needed values.
Besides, with the current DeepPavlov's functionality you have to recreate the object anyway (consecutive training and interact modes recreate objects).